### PR TITLE
support alternate aws partitions

### PR DESCRIPTION
**Pull request by woz5999**
## what
- dynamically set the aws partition when constructing ARNs

## why
- hardcoded aws partitions make govcloud users sad

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -3,6 +3,7 @@ locals {
 }
 
 data "aws_caller_identity" "current" {}
+data "aws_partition" "current" {}
```
Comment (nitrocode) [inline main.tf]: Can you use count logic here to disable these data sources too ?

File: ssm_log_bucket.tf [hunk 2]
```diff
@@ -1,5 +1,6 @@
 locals {
   account_id        = data.aws_caller_identity.current.account_id
+  aws_partition     = data.aws_partition.current.partition
```
Comment (nitrocode) [inline ssm_log_bucket.tf]: After count logic is applied

```suggestion
  account_id        = join("", data.aws_caller_identity.current.*.account_id)
  aws_partition     = join("", data.aws_partition.current.*.partition)
```

**Comment by nitrocode** [comment]
Thanks for the contribution @woz5999 ! Please see comments

**Comment by woz5999** [comment]
@nitrocode fixed

**Review by nitrocode — APPROVED** [review]

**Pull request merged**